lib/code_writer.rb

```diff
@@ -121,4 +102,37 @@ def write_arithmetic(command)
   def close
     @out.close
   end
+
+  private
+
+  def get_memory_address_for_pop(segment, index)
```

Comment:
This is definitely a better name. FWIW I think you could go even further and say `#get_address_for_pop` — the “memory” is redundant because “address” already implies it.